src/notification/email/EMailNotificationTask.ts

```diff
@@ -1,48 +1,51 @@
+import { Message, SMTPClient, SMTPError } from 'emailjs';
 
 import BackendError from '../../exception/BackendError';
 import Configuration from '../../utils/Configuration';
 import Constants from '../../utils/Constants';
+import EmailConfiguration from '../../types/configuration/EmailConfiguration';
 import Logging from '../../utils/Logging';
 import NotificationHandler from '../NotificationHandler';
 import NotificationTask from '../NotificationTask';
-import { SMTPClient } from 'emailjs';
 import { ServerAction } from '../../types/Server';
 import Tenant from '../../types/Tenant';
 import User from '../../types/User';
 import Utils from '../../utils/Utils';
 import ejs from 'ejs';
 import fs from 'fs';
 import global from '../../types/GlobalType';
+import rfc2047 from 'rfc2047';
 
 const MODULE_NAME = 'EMailNotificationTask';
 
 export default class EMailNotificationTask implements NotificationTask {
-  private server: any;
-  private serverBackup: any;
-  private emailConfig = Configuration.getEmailConfig();
-
-  constructor() {
-    // Connect the SMTP server
-    this.server = new SMTPClient({
-      user: this.emailConfig.smtp.user,
-      password: this.emailConfig.smtp.password,
-      host: this.emailConfig.smtp.host,
-      port: this.emailConfig.smtp.port,
-      tls: this.emailConfig.smtp.requireTLS,
-      ssl: this.emailConfig.smtp.secure
-    });
-    // Connect the SMTP Backup server
-    if (this.emailConfig.smtpBackup) {
-      this.serverBackup = new SMTPClient({
-        user: this.emailConfig.smtpBackup.user,
-        password: this.emailConfig.smtpBackup.password,
-        host: this.emailConfig.smtpBackup.host,
-        port: this.emailConfig.smtpBackup.port,
-        tls: this.emailConfig.smtpBackup.requireTLS,
-        ssl: this.emailConfig.smtpBackup.secure
+  private static instance: EMailNotificationTask;
+  private emailConfig: EmailConfiguration = Configuration.getEmailConfig();
+  private SMTPMainClientInstance: SMTPClient;
+  private SMTPBackupClientInstance: SMTPClient;
+  private backupInUse: boolean;
+
+  private constructor() {
+    // Connect to the SMTP servers
+    if (!Utils.isUndefined(this.emailConfig.smtp)) {
+      this.SMTPMainClientInstance = new SMTPClient({
+        user: this.emailConfig.smtp.user,
+        password: this.emailConfig.smtp.password,
+        host: this.emailConfig.smtp.host,
+        port: this.emailConfig.smtp.port,
+        tls: this.emailConfig.smtp.requireTLS,
+        ssl: this.emailConfig.smtp.secure
       });
     }
+    this.backupInUse = false;
```

Comment:
Create the SMTPBackupClientInstance here also.
Remove backupInUse member variable.

Reply:
The implementation in the PR only create the SMTP client instance for the backup server in case of an error on the main SMTP server. 
The code that do that small optimisation is easy to read and understand and I do not see why it should be replaced by a less optimised and readable code with syntactic sugar.